flipperdevices/flipperzero-good-faps
# [FL-3773] VGM Tool: Fix crashes

# What's new

- Fix crash due to logging output in critical section
- Fix crash if the app is built with `DEBUG=1`
- Bump version to 1.2

# Verification 

1. Install the app.
2. Go to `Settings -> System` and set `LogLevel` to `Default` or `Info`.
3. Run the app. It does not matter whether the VGM is connected or not. The app should not crash regardless of the `LogLevel` setting.
4. Connect the Video Game Module (if it is not already connected) and ensure that the app is functioning as intended and no regressions are present.

# Caveats
- If built with `DEBUG=1` (for example, with `ufbt` on dev SDK), the flashing process is considerably slower. Use `DEBUG=0 COMPACT=1` to achieve nominal performance.

# Checklist (For Reviewer)
